### PR TITLE
Fix content.integration tests warnings

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

- Fixes most of the warnings in the Content.IntegrationTests module.
- Does not fix CreateGrid related warnings as that would require engine work.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs [hunk 1]
```diff
@@ -65,18 +68,24 @@ public async Task TestDeletedCanReconnect()
         // Delete entity
         Assert.That(entMan.EntityExists(entity));
         await pair.Server.WaitPost(() => entMan.DeleteEntity(entity));
-        Assert.That(entMan.Deleted(entity));
-        Assert.IsNull(mind.OwnedEntity);
+        Assert.Multiple(() =>
+        {
+            Assert.That(entMan.Deleted(entity));
+            Assert.That(mind.OwnedEntity, Is.Not.Null);
```
Comment (TemporalOroboros) [inline Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs:74]: Should be Is.Null

File: Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs [hunk 2]
```diff
@@ -134,11 +146,14 @@ public async Task TestVisitingReconnect()
...
-        Assert.That(!entMan.Deleted(visiting));
-        Assert.That(mind.CurrentEntity == visiting);
-        Assert.That(mind.CurrentEntity == visiting);
+        Assert.Multiple(() =>
+        {
+            Assert.That(GetMind(pair), Is.EqualTo(mind));
+            Assert.That(!entMan.Deleted(original), Is.False);
+            Assert.That(!entMan.Deleted(visiting), Is.False);
+            Assert.That(mind.CurrentEntity, Is.EqualTo(visiting));
+            Assert.That(mind.CurrentEntity, Is.EqualTo(visiting));
```
Comment (TemporalOroboros) [inline Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs:155]: Probably don't need two of the same test?

File: Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs [hunk 3]
```diff
@@ -134,11 +146,14 @@ public async Task TestVisitingReconnect()
...
-        Assert.That(mind == GetMind(pair));
-        Assert.That(!entMan.Deleted(original));
-        Assert.That(!entMan.Deleted(visiting));
-        Assert.That(mind.CurrentEntity == visiting);
-        Assert.That(mind.CurrentEntity == visiting);
+        Assert.Multiple(() =>
+        {
+            Assert.That(GetMind(pair), Is.EqualTo(mind));
+            Assert.That(!entMan.Deleted(original), Is.False);
+            Assert.That(!entMan.Deleted(visiting), Is.False);
```
Comment (TemporalOroboros) [inline Content.IntegrationTests/Tests/Minds/MindTests.ReconnectTests.cs:153]: Both of the checks here shouldn't have the !s. That's what the Is.False s are for.

File: Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs [hunk 4]
```diff
@@ -136,24 +136,27 @@ public async Task Test()
 
                     // Event got repeated on client as a past prediction.
                     Assert.That(clientSystem.EventTriggerList,
```
Comment (TemporalOroboros) [inline Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs:138]: Somehow got here as False, True, True, True. Fix.

**Review by TemporalOroboros — COMMENTED** [review]
Broke some tests. Will fix them soon.

File: Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs (hunk 4 above)
Comment (TemporalOroboros) [reply Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs:138]: Might be fixed by dirtying the comp after handling state.

Comment (TemporalOroboros) [reply Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs:138]: Did not fix it. Not sure what's wrong here.

Comment (TemporalOroboros) [reply Content.IntegrationTests/Tests/Networking/SimplePredictReconcileTest.cs:138]: Looks like the component state isn't being reset during prediction? I don't know why that's happening.

**Comment by TemporalOroboros** [comment]
Should be good to go. About half of the lines added are two new mispredict reconciliation tests. They're for the source gen and ECS based versions.

File: Content.IntegrationTests/PoolManager.cs [hunk 5]
```diff
@@ -108,9 +104,15 @@ public static class PoolManager
...
+                .LoadExtraSystemType<AutoPredictReconcileTest.AutoPredictionTestEntitySystem>();
+            IoCManager.Resolve<IComponentFactory>().RegisterClass<AutoPredictionTestComponent>();
             IoCManager.Resolve<IEntitySystemManager>()
                 .LoadExtraSystemType<SimplePredictReconcileTest.PredictionTestEntitySystem>();
             IoCManager.Resolve<IComponentFactory>().RegisterClass<SimplePredictReconcileTest.PredictionTestComponent>();
+            IoCManager.Resolve<IEntitySystemManager>()
+                .LoadExtraSystemType<SystemPredictReconcileTest.SystemPredictionTestEntitySystem>();
+            IoCManager.Resolve<IComponentFactory>().RegisterClass<SystemPredictReconcileTest.SystemPredictionTestComponent>();
             IoCManager.Register<ResettingEntitySystemTests.TestRoundRestartCleanupEvent>();
             IoCManager.Register<InteractionSystemTests.TestInteractionSystem>();
```
Comment (metalgearsloth) [inline Content.IntegrationTests/PoolManager.cs:117]: Just do the resolve once and re-use it.

File: Content.IntegrationTests/PoolManager.cs [hunk 6]
```diff
@@ -211,8 +213,16 @@ public static string DeathReport()
...
+                        .LoadExtraSystemType<AutoPredictReconcileTest.AutoPredictionTestEntitySystem>();
+                    IoCManager.Resolve<IComponentFactory>()
+                        .RegisterClass<AutoPredictionTestComponent>();
                     IoCManager.Resolve<IEntitySystemManager>()
                         .LoadExtraSystemType<SimplePredictReconcileTest.PredictionTestEntitySystem>();
+                    IoCManager.Resolve<IEntitySystemManager>()
+                        .LoadExtraSystemType<SystemPredictReconcileTest.SystemPredictionTestEntitySystem>();
+                    IoCManager.Resolve<IComponentFactory>()
+                        .RegisterClass<SystemPredictReconcileTest.SystemPredictionTestComponent>();
                     IoCManager.Resolve<IComponentFactory>()
```
Comment (metalgearsloth) [inline Content.IntegrationTests/PoolManager.cs:226]: Same here.

File: Content.IntegrationTests/PoolManager.cs [hunk 7]
```diff
@@ -573,7 +589,10 @@ public static async Task<TestMapData> CreateTestMap(PairTracker pairTracker)
             mapData.MapId = mapManager.CreateMap();
             mapData.MapUid = mapManager.GetMapEntityId(mapData.MapId);
             mapData.MapGrid = mapManager.CreateGrid(mapData.MapId);
-            mapData.GridUid = mapData.MapGrid.Owner;
+            // The API does not provide a way to get the Uid of the above grid other than this.
+            mapData.GridUid = entityManager.GetComponent<TransformComponent>(mapData.MapUid)
+                .ChildEntities
+                .First((ent) => entityManager.GetComponent<MapGridComponent>(ent) == mapData.MapGrid);
```
Comment (metalgearsloth) [inline Content.IntegrationTests/PoolManager.cs:595]: Better to leave the warning in place than this.

File: Content.IntegrationTests/PoolManager.cs [hunk 8]
```diff
@@ -809,7 +828,10 @@ public bool CanFastRecycle(PoolSettings nextSettings)
     // Prototype hot reload is not available outside TOOLS builds,
     // so we can't pool test instances that use ExtraPrototypes without TOOLS.
 #if TOOLS
+    // Shuts up the warning that Omnisharp throws b/c it can't see both branches of the if/else/endif.
+#pragma warning disable
     private bool NoToolsExtraPrototypes => false;
+#pragma warning restore
```
Comment (metalgearsloth) [inline Content.IntegrationTests/PoolManager.cs:834]: Should just disable the specific warning rather than all of them.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**